Mathlib/Analysis/ODE/PicardLindelof.lean

```diff
@@ -391,79 +391,50 @@ theorem IsPicardLindelof.exists_forall_hasDerivWithinAt_Icc_eq [CompleteSpace E]
-  -- extract Lipschitz constant
-  obtain ⟨L, s', hs', hlip⟩ :=
-    ContDiffAt.exists_lipschitzOnWith ((hv.contDiffWithinAt (mem_of_mem_nhds hs)).contDiffAt hs)
-  -- radius of closed ball in which v is bounded
-  obtain ⟨r, hr : 0 < r, hball⟩ := Metric.mem_nhds_iff.mp (inter_sets (𝓝 x₀) hs hs')
-  have hr' := (half_pos hr).le
-  -- uses [ProperSpace E] for `isCompact_closedBall`
-  obtain ⟨C, hC⟩ := (isCompact_closedBall x₀ (r / 2)).bddAbove_image <| hv.continuousOn.norm.mono
-    (subset_inter_iff.mp ((closedBall_subset_ball (half_lt_self hr)).trans hball)).left
-  have hC' : 0 ≤ C := by
-    apply (norm_nonneg (v x₀)).trans
-    apply hC
-    exact ⟨x₀, ⟨mem_closedBall_self hr', rfl⟩⟩
-  set ε := if C = 0 then 1 else r / 2 / C with hε
-  have hε0 : 0 < ε := by
-    rw [hε]
-    split_ifs with h
-    · exact zero_lt_one
-    · exact div_pos (half_pos hr) (lt_of_le_of_ne hC' (Ne.symm h))
-  refine' ⟨ε, hε0, L, r / 2, C, _⟩
+theorem exists_isPicardLindelof_const_of_contDiffAt (hv : ContDiffAt ℝ 1 v x₀) :
+  ∃ ε > (0 : ℝ), ∃ L R C, IsPicardLindelof (fun _ => v) (t₀ - ε) t₀ (t₀ + ε) x₀ L R C := by
+  obtain ⟨L, s, hs, hlip⟩ := hv.exists_lipschitzOnWith
+  obtain ⟨R₁, hR₁ : 0 < R₁, hball⟩ := Metric.mem_nhds_iff.mp hs
+  obtain ⟨R₂, hR₂ : 0 < R₂, hbdd⟩ := Metric.continuousAt_iff.mp hv.continuousAt.norm 1 zero_lt_one
+  have hbdd' : ∀ x ∈ Metric.ball x₀ R₂, ‖v x‖ ≤ 1 + ‖v x₀‖ := fun _ hx =>
+    sub_le_iff_le_add.mp <| le_of_lt <| lt_of_abs_lt <| Real.dist_eq _ _ ▸ hbdd hx
+  set ε := min R₁ R₂ / 2 / (1 + ‖v x₀‖) with hε
+  have hε0 : 0 < ε := hε ▸ div_pos (half_pos <| lt_min hR₁ hR₂)
+    (add_pos_of_pos_of_nonneg zero_lt_one (norm_nonneg _))
+  refine' ⟨ε, hε0, L, min R₁ R₂ / 2, 1 + ‖v x₀‖, _⟩
   exact
-    { ht₀ := by rw [← Real.closedBall_eq_Icc]; exact mem_closedBall_self hε0.le
-      hR := (half_pos hr).le
-      lipschitz := fun t _ => hlip.mono
-        (subset_inter_iff.mp (Subset.trans (closedBall_subset_ball (half_lt_self hr)) hball)).2
-      cont := fun x _ => continuousOn_const
-      norm_le := fun t _ x hx => hC ⟨x, hx, rfl⟩
+    { ht₀ := Real.closedBall_eq_Icc ▸ mem_closedBall_self hε0.le
+      hR := le_of_lt <| half_pos <| lt_min hR₁ hR₂
+      lipschitz := fun _ _ => hlip.mono <| Subset.trans
+        (closedBall_subset_ball <| half_lt_self <| lt_min hR₁ hR₂)
+        (Subset.trans (Metric.ball_subset_ball <| min_le_left _ _) hball)
+      cont := fun _ _ => continuousOn_const
+      norm_le := fun _ _ x hx => hbdd' x <| mem_of_mem_of_subset hx <| Subset.trans
+        (closedBall_subset_ball <| half_lt_self <| lt_min hR₁ hR₂)
+        (Subset.trans (Metric.ball_subset_ball <| min_le_right _ _) (subset_refl _))
       C_mul_le_R := by
-        rw [add_sub_cancel', sub_sub_cancel, max_self, mul_ite, mul_one]
-        split_ifs with h
-        · rwa [← h] at hr'
-        · exact (mul_div_cancel' (r / 2) h).le }
-#align exists_is_picard_lindelof_const_of_cont_diff_on_nhds exists_isPicardLindelof_const_of_contDiffOn_nhds
```

Comment:
You should not delete #aligns (same below)

Reply:
What if the theorem name and statement have changed?

Reply:
I would still keep the `#align`.